pyintegration/deephaven2/time.py

```diff
@@ -0,0 +1,775 @@
+    """ Returns an 1-based int value of the day of the year (Julian date) for a DateTime in the specified time zone.
+
+    Args:
+        dt (DateTime): the DateTime for which to find the day of the year
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.dayOfYear(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def diff_nanos(dt1: DateTime, dt2: DateTime) -> int:
+    """ Returns the difference in nanoseconds between two DateTime values.
+
+    Args:
+        dt1 (DateTime): the 1st DateTime
+        dt2 (DateTime): the 2nd DateTime
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.diffNanos(dt1, dt2)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def format_datetime(dt, tz: TimeZone) -> str:
+    """ Returns a string date/time representation formatted as yyyy-MM-ddThh:mm:ss.nnnnnnnnn TZ.
+
+    Args:
+        dt (DateTime): the DateTime to format as a string
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        str
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.format(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def format_nanos(ns: int) -> str:
+    """ Returns a string date/time representation formatted as yyyy-MM-ddThh:mm:ss.nnnnnnnnn.
+
+    Args:
+        ns (int): the number of nanoseconds from the Epoch
+
+    Returns:
+        str
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.format(ns)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def format_date(dt, tz: TimeZone) -> str:
+    """ Returns a string date representation of a DateTime interpreted for a specified time zone formatted as yyy-MM-dd.
+
+    Args:
+        dt (DateTime): the DateTime to format
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        str
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.formatDate(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def hour_of_day(dt, tz: TimeZone) -> int:
+    """ Returns an int value of the hour of the day for a DateTime in the specified time zone. The hour is on a
+     24 hour clock (0 - 23).
+
+    Args:
+        dt (DateTime): the DateTime for which to find the hour of the day
+        tz (TimeZone): the TimeZone to use when interpreting the date/time
+
+    Returns:
+        int
+
+    Raises:
+        DHError
+    """
+    try:
+        return _JDateTimeUtils.hourOfDay(dt, tz.value)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def is_after(dt1: DateTime, dt2: DateTime) -> bool:
+    """ Evaluates whether one DateTime value is later than a second DateTime value.
```

Comment:
In the docs `date/time`, `datetime`, and `DateTime` are used.  It seems like we probably want to use `DateTime` most places.  The existing docs use a hyphenated `date-time` when not using the specific `DateTime`.
https://deephaven.io/core/docs/how-to-guides/work-with-date-time/

I also see in that link that we are using `f` for fractions of a second.  Yet another variation.